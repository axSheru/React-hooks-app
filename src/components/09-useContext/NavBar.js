import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    //NOTE: La diferencia entre Link y NavLink es que Navlink nos permite aplicar CSS si estamos en esa ruta.
    //NOTE: activeClassName se usa en versiones previas a V6 de React-Router-DOM
    //<NavLink activeClassName="active" className="nav-link" aria-current="page" to="/">Home</NavLink>
    //NOTE: Se usa una funci√≥n dentro de className para definir el active condicional en V6 y posteriores.
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link className="navbar-brand" to="/">useContext</Link>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink className={`nav-link ${(navData) => navData.isActive && 'active'}`} aria-current="page" to="/">Home</NavLink>
                        <NavLink className={`nav-link ${(navData) => navData.isActive && 'active'}`} to="/about">About</NavLink>
                        <NavLink className={`nav-link ${(navData) => navData.isActive && 'active'}`} to="/login">Login</NavLink>
                    </div>
                </div>
            </div>
        </nav>
    );
}
