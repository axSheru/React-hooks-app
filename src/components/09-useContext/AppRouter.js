import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Routes,
  Navigate
} from "react-router-dom";

import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className="container">
                    <Routes>
                        // NOTE: After V6 by default exact is defined, this is to match exactly the given route.
                        <Route exact path='/' element={ <HomeScreen /> } />
                        <Route path='/about' element={ <AboutScreen /> } />
                        <Route path='/login' element={ <LoginScreen /> } />
                        <Route path='*' element={ <Navigate to="/" /> } />
                    </Routes>
                </div>
            </div>
        </Router>
    );
}
