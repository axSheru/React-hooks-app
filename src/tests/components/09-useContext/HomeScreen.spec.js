import { mount } from "enzyme";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en el componente HomeScreen.', () => {

    const user = {
        name: 'Alex',
        email: 'alex@test.com'
    };

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('debe de hacer match con el snapshot.', () => {

        expect( wrapper ).toMatchSnapshot();

    });

});