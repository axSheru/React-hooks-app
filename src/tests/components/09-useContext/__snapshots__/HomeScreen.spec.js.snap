// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente HomeScreen. debe de hacer match con el snapshot. 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "name": "Alex",
   "email": "alex@test.com"
}
  </pre>
</div>
`;
